packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -1091,6 +1090,37 @@ export default class VTable {
             handler(currentCell.td);
         }
     }
+
+    /**
+     * Transforms the selected cells to Ranges.
+     * For Each Row a Range with selected cells, a Range is going to be returned.
+     * @returns
+     */
+    getSelectedRanges() {
+        const ranges: Range[] = [];
+        const rows = this.cells.length;
+
+        for (let y = 0; y < rows; y++) {
+            const rowRange = new Range();
+            let firstSelected: HTMLTableCellElement = null;
+            let lastSelected: HTMLTableCellElement = null;
+
+            this.forEachCellOfRow(y, (cell, x) => {
+                if (cell.td && this.isInsideOfSelection(x, y)) {
+                    firstSelected = firstSelected || cell.td;
+                    lastSelected = cell.td;
+                }
+            });
+
+            if (firstSelected) {
+                rowRange.setStartBefore(firstSelected);
+                rowRange.setEndAfter(lastSelected);
+                ranges.push(rowRange);
+            }
+        }
+
+        return ranges.length > 0 ? ranges : null;

```

Comment:
I prefer return empty array when nothing selected. So that we don't need always check null when use the return value

Reply:
Ok, that makes sense!